trestle/core/validator_helper.py

```diff
@@ -150,35 +66,7 @@ def has_no_duplicate_values_by_name(object_of_interest: Any, name_of_interest: s
     return False
 
 
-def find_all_attribs_by_regex(object_of_interest: Any, regex_of_interest: str) -> List[Tuple[str, Any]]:
-    """Find all attributes in object matching regex expression."""
-    all_attrs = []
-    p = re.compile(regex_of_interest)
-    if isinstance(object_of_interest, pydantic.BaseModel):
-        # fields_set has names of fields set when model was initialized
-        fields = getattr(object_of_interest, '__fields_set__', None)
-        for field in fields:
-            if p.findall(field):
-                all_attrs.append((field, object_of_interest.__dict__[field]))
-            new_attrs = find_all_attribs_by_regex(object_of_interest.__dict__[field], regex_of_interest)
-            all_attrs.extend(new_attrs)
-        return all_attrs
-    if type(object_of_interest) is list:
-        for item in object_of_interest:
-            new_attrs = find_all_attribs_by_regex(item, regex_of_interest)
-            all_attrs.extend(new_attrs)
-        return all_attrs
-    if type(object_of_interest) is dict:
-        for key, value in object_of_interest.items():
-            if p.findall(key):
-                all_attrs.append((key, value))
-            new_attrs = find_all_attribs_by_regex(value, regex_of_interest)
-            all_attrs.extend(new_attrs)
-        return all_attrs
-    return all_attrs
-
-
-def regenerate_uuids_in_place(object_of_interest: Any, uuid_lut: Dict[str, str]) -> Tuple[Any, Dict[str, str]]:
+def _regenerate_uuids_in_place(object_of_interest: Any, uuid_lut: Dict[str, str]) -> Tuple[Any, Dict[str, str]]:
```

Comment:
This function contains a [code smell](https://sonarcloud.io/project/issues?id=compliance-trestle&open=AX5rlNIt7Nfw18hwjlm2&pullRequest=993&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL) do you think we can simplify it?